src/common.rs

```diff
@@ -1,3 +1,45 @@
+/// A CPU architecture.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
+pub enum Architecture {
+    Unknown,
+    Aarch64,
+    Arm,
+    I386,
+    Mips,
+    Wasm,
+    X86_64,
+}
+
+/// A binary file format.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
+pub enum BinaryFormat {
+    Coff,
+    Elf,
+    MachO,
+    Pe,
+    Wasm,
+}
+
+/// The width of a pointer value in the file format.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[allow(missing_docs)]
+pub enum PointerWidth {
```

Comment:
Make this `#[repr(u8)]` and then rather than matching in the `bytes` method below, just do `self as u8`?

Reply:
Nice, I didn't know about that.